app/src/main/java/com/example/socialmediavbsanalay/presentation/fragments/MainPageFragment.kt

```diff
@@ -0,0 +1,196 @@
+
+import android.app.AlertDialog
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.provider.MediaStore
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.socialmediavbsanalay.R
+import com.example.socialmediavbsanalay.databinding.FragmentMainPageBinding
+import com.example.socialmediavbsanalay.presentation.adapters.PostAdapter
+import com.example.socialmediavbsanalay.presentation.adapters.StoryAdapter
+import dagger.hilt.android.AndroidEntryPoint
+import android.Manifest
+import android.content.Context
+import androidx.activity.result.contract.ActivityResultContracts
+
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [MainPageFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+@AndroidEntryPoint
+class MainPageFragment : Fragment() {
+
+    private var _binding: FragmentMainPageBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var storyAdapter: StoryAdapter
+    private lateinit var postAdapter: PostAdapter
+    private val REQUEST_IMAGE_CAPTURE = 1
+    private val PICK_IMAGE = 2
+    private val REQUEST_CODE_READ_EXTERNAL_STORAGE = 100
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted: Boolean ->
+        if (isGranted) {
+            accessGallery()
+        } else {
+            // İzin reddedildiğinde kullanıcıya bilgilendirme yap
+            Toast.makeText(requireContext(), "İzin gerekli!", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun accessGallery() {
+        // Galeriden fotoğraf çekme işlemleri burada yapılacak
+        // MediaStore veya Intent kullanarak galeriye erişebilirsiniz
+        val galleryIntent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+        startActivityForResult(galleryIntent, GALLERY_REQUEST_CODE)
+    }
+
+    companion object {
+        private const val GALLERY_REQUEST_CODE = 123
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentMainPageBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        storyAdapter = StoryAdapter()
```

Comment:
adapterların oluşturulduğu kodları ve clickListenerların setlediği kodları ayrı bir fonksiyon oluşturup ( örneğin initRecyclerViews() ve initClickListeners() ) o fonksiyonların içine taşır mısın? Kod olarak bir yanlışlık yok ama daha clean gözükmesi açısından bu şekilde kullanılabilir.

Reply:
Halledildi MainPage fragmentle ilgili permisinlarını orada vermiştim ortadaki butona basınca oradan ilerliyordu ancak bunu mainPageFragmeninden yönetirsem diğer fragmentlerde post atamam main de permisinlarını vermeye çalıştığımda da hata aldım